### PR TITLE
Add Project documentation

### PR DESCRIPTION
**Pull request by sergiomsantos**
This PR adds documentation for the `Project` API.

### CODE REVIEW

File: tutorials/intro_to_api/projects.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+# task2 will be added to the default project
+task2 = simulator.run(input_dir=input_dir,
+                      sim_config_filename="params.txt")
+
+print(task1.get_info()['project']) # "my_xbeach_project"
+print(task2.get_info()['project']) # "userab1cdef2" (default project)
+```
+
+#### Explicit management
```
Comment (sarmento) [inline tutorials/intro_to_api/projects.md:129]: Maybe we should start with the explicit option.

**Review by sarmento — COMMENTED** [review]
This looks great!
Thank you!
I wonder if we really need the append=True. Should that be actually a permanent state? Why? Why not?
Can you pls clarify that somewhere?

**Review by sarmento — COMMENTED** [review]
LGTM!

File: tutorials/intro_to_api/projects.md (hunk 1 above)
Comment (sergiomsantos) [reply tutorials/intro_to_api/projects.md:129]: Done in the development branch.

**Comment by sergiomsantos** [comment]
> This looks great! Thank you! I wonder if we really need the append=True. Should that be actually a permanent state? Why? Why not? Can you pls clarify that somewhere?

The only reason to have this behaviour is so that the user is aware that tasks will be appended to a project if a project is open. It doesn't need to be that way, but at least it provides some sort of security check to ensure that the user is aware of what is going on. This is explained in the `Adding a Task to a Project` section, right at the top.
We can change this behaviour, though.

**Pull request merged**